core/engine/src/module/source.rs

```diff
@@ -1763,21 +1763,27 @@ impl SourceTextModule {
         // 6. Set the VariableEnvironment of moduleContext to module.[[Environment]].
         // 7. Set the LexicalEnvironment of moduleContext to module.[[Environment]].
         let env_fp = environments.len() as u32;
-        let mut callframe = CallFrame::new(
+        let callframe = CallFrame::new(
             codeblock,
             Some(ActiveRunnable::Module(self.parent())),
             environments,
             realm,
         )
         .with_env_fp(env_fp)
         .with_flags(CallFrameFlags::EXIT_EARLY);
-        callframe.promise_capability = capability;
 
         // 8. Suspend the running execution context.
         context
             .vm
             .push_frame_with_stack(callframe, JsValue::undefined(), JsValue::null());
 
+        context
+            .vm
+            .frames
+            .last()
+            .expect("there should be a frame")
+            .set_promise_capability(&mut context.vm.stack, capability);
```

Comment:
Thought: Seeing this like this makes me wonder if we should start migrating to a register-first architecture, where we have a certain amount of utility registers on Vm so that the callee can share data with the caller and viceversa.